hie-build-plugin/Haskell/Ide/BuildPlugin.hs

```diff
@@ -216,16 +302,31 @@ flagToJSON f = object ["name" .= ((\(FlagName s) -> s) $ flagName f), "descripti
 
 -----------------------------------------------
 
-buildDirectory :: CommandFunc ()
-buildDirectory = CmdSync $ \ctx req -> withCommonArgs ctx req $ do
+data BuildParams = BP {
+         -- common params. horrible
```

Comment:
Perhaps we should merge it as is, and then clean it up?

The intent here was to get it into the new architecture.